op-challenger/fault/agent.go

```diff
@@ -65,12 +71,32 @@ func (a *Agent) Act(ctx context.Context) error {
 	return nil
 }
 
+// ShouldResolve returns true if the agent should resolve the game.
+func (a *Agent) ShouldResolve(ctx context.Context) bool {
+	status, err := a.statusFetcher.GetGameStatus(ctx)
```

Comment:
Why would the status be `types.GameStatusDefenderWon` or `types.GameStatusChallengerWon`  before the resolve function has been called?

Reply:
Another challenger could have resolved the game?

Reply:
Ah I see why you're asking - addressing...